File: wavelength-api/Domain/GameSession.cs
namespace Domain;

public class GameSession
{
    public Guid Id { get; set; }

    public DateTime? StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public string JoinCode { get; set; }

    public Guid OwnerId { get; set; }

    public int GameRound { get; set; }

    /// <summary>
    ///     Gets or sets the object that holds the score for each team for this game.
    /// </summary>
    public GameSessionResult GameSessionResult { get; set; }

    public ICollection<GameSessionMember> Members { get; set; } = new List<GameSessionMember>();

    public ICollection<GameRound> Rounds { get; set; } = new List<GameRound>();
}

File: wavelength/src/models/GameSession.ts
import { GameRound } from "./GameRound";
import GameSessionMember from "./GameSessionMember";
import { GameSessionResult } from "./GameSessionResult";

type GameSession = {
  id: string;
  startTime?: Date;
  endTime?: Date;
  joinCode: string;
  ownerId: string;
  gameRound: number;
  members: GameSessionMember[];
  rounds: GameRound[];
  scores: GameSessionResult;
};

export default GameSession;

export function getCurrentGameRound(gameSession: GameSession | undefined) {
  if (
    gameSession &&
    gameSession.gameRound >= 0 &&
    gameSession.rounds.length > gameSession.gameRound
  )
    return gameSession.rounds[gameSession?.gameRound];

  return undefined;
}
